server/src/domain/domain.constant.spec.ts

```diff
@@ -0,0 +1,169 @@
+    { mimetype: 'image/raf', extension: 'raf' },
+    { mimetype: 'image/raw', extension: 'raw' },
+    { mimetype: 'image/rwl', extension: 'rwl' },
+    { mimetype: 'image/sr2', extension: 'sr2' },
+    { mimetype: 'image/srf', extension: 'srf' },
+    { mimetype: 'image/srw', extension: 'srw' },
+    { mimetype: 'image/tiff', extension: 'tiff' },
+    { mimetype: 'image/webp', extension: 'webp' },
+    { mimetype: 'image/x-adobe-dng', extension: 'dng' },
+    { mimetype: 'image/x-arriflex-ari', extension: 'ari' },
+    { mimetype: 'image/x-canon-cr2', extension: 'cr2' },
+    { mimetype: 'image/x-canon-cr3', extension: 'cr3' },
+    { mimetype: 'image/x-canon-crw', extension: 'crw' },
+    { mimetype: 'image/x-epson-erf', extension: 'erf' },
+    { mimetype: 'image/x-fuji-raf', extension: 'raf' },
+    { mimetype: 'image/x-hasselblad-3fr', extension: '3fr' },
+    { mimetype: 'image/x-hasselblad-fff', extension: 'fff' },
+    { mimetype: 'image/x-kodak-dcr', extension: 'dcr' },
+    { mimetype: 'image/x-kodak-k25', extension: 'k25' },
+    { mimetype: 'image/x-kodak-kdc', extension: 'kdc' },
+    { mimetype: 'image/x-leica-rwl', extension: 'rwl' },
+    { mimetype: 'image/x-minolta-mrw', extension: 'mrw' },
+    { mimetype: 'image/x-nikon-nef', extension: 'nef' },
+    { mimetype: 'image/x-olympus-orf', extension: 'orf' },
+    { mimetype: 'image/x-olympus-ori', extension: 'ori' },
+    { mimetype: 'image/x-panasonic-raw', extension: 'raw' },
+    { mimetype: 'image/x-pentax-pef', extension: 'pef' },
+    { mimetype: 'image/x-phantom-cin', extension: 'cin' },
+    { mimetype: 'image/x-phaseone-cap', extension: 'cap' },
+    { mimetype: 'image/x-phaseone-iiq', extension: 'iiq' },
+    { mimetype: 'image/x-samsung-srw', extension: 'srw' },
+    { mimetype: 'image/x-sigma-x3f', extension: 'x3f' },
+    { mimetype: 'image/x-sony-arw', extension: 'arw' },
+    { mimetype: 'image/x-sony-sr2', extension: 'sr2' },
+    { mimetype: 'image/x-sony-srf', extension: 'srf' },
+    { mimetype: 'image/x3f', extension: 'x3f' },
+    { mimetype: 'video/3gpp', extension: '3gp' },
+    { mimetype: 'video/avi', extension: 'avi' },
+    { mimetype: 'video/mp2t', extension: 'm2ts' },
+    { mimetype: 'video/mp2t', extension: 'mts' },
+    { mimetype: 'video/mp4', extension: 'mp4' },
+    { mimetype: 'video/mpeg', extension: 'mpg' },
+    { mimetype: 'video/msvideo', extension: 'avi' },
+    { mimetype: 'video/quicktime', extension: 'mov' },
+    { mimetype: 'video/vnd.avi', extension: 'avi' },
+    { mimetype: 'video/webm', extension: 'webm' },
+    { mimetype: 'video/x-flv', extension: 'flv' },
+    { mimetype: 'video/x-matroska', extension: 'mkv' },
+    { mimetype: 'video/x-ms-wmv', extension: 'wmv' },
+    { mimetype: 'video/x-msvideo', extension: 'avi' },
+  ]) {
+    it(`should map ${extension} to ${mimetype}`, async () => {
+      expect({ ...mimeTypes.image, ...mimeTypes.video }[`.${extension}`]).toContain(mimetype);
```

Comment:
You are reconstructing this object every test. Just do it once outside of the for loop.

Reply:
Can we keep extensions the same here? Or add the period in front of the extension in the list

Reply:
>Can we keep extensions the same here?

What do you mean by this? I understand the second option, but not this one.

Reply:
Also, we are extension driven now, not mime-type driven, so I think it makes more sense to have extension be the first key and mime type be the last key, and then it would make sense for this to be sorted by extension name not mime type.

Reply:
> Also, we are extension driven now, not mime-type driven, so I think it makes more sense to have extension be the first key and mime type be the last key, and then it would make sense for this to be sorted by extension name not mime type.

This is tough - I think sorting by extensions makes sense when we have `extension: [mimeType]` but for `{ mimeType, extension }` it makes the most sense to sort by mimeType as to keep the grouping.

Reply:
> You are reconstructing this object every test. Just do it once outside of the for loop.

I do get this, though this is a test and is unlikely to have any impact on performance. Having to define some constant elsewhere just because feels weird.

Reply:
Right now if you want to add a new file extension you need to:
- add it to the mapping with a mime type
- update the asset upload test to include the extension
- And now go here and also update this list, which isn't even sorted in the same way. 

I don't know why it needs to be so complicated. Why do we need to validate all variations of mime types, why do we need it sorted alphabetically in the first place? Why can't the test be validating the actual lookup function itself, which you still aren't even testing. I think it is confusing to have this be mime type driven when everything else is extension driven.